pkgs/development/compilers/zig/setup-hook.sh

```diff
@@ -0,0 +1,74 @@
+# shellcheck shell=bash
+
+function zigSetGlobalCacheDir {
+
+    export ZIG_GLOBAL_CACHE_DIR=$(mktemp -d)
+}
+
+function zigBuildPhase {
+    runHook preBuild
+
+    # The local zigDefaultBuildFlags flagset below is meant to avoid CPU feature
+    # impurity in Nixpkgs. However, this flagset is "unstable", because it is
+    # not always exposed by the developer. Because of this, @andrewrk proposed
+    # some solutions that will be implemented in future releases of Zig.
+
+    # Below are some useful links describing the discovery process of this
+    # 'bug' in Nixpkgs:
+
+    # https://github.com/NixOS/nixpkgs/issues/169461
+    # https://github.com/NixOS/nixpkgs/issues/185644
+    # https://github.com/NixOS/nixpkgs/pull/197046
+    # https://github.com/NixOS/nixpkgs/pull/241741#issuecomment-1624227485
+    # https://github.com/ziglang/zig/issues/14281#issuecomment-1624220653
+
+    # Many thanks to the Zig Foundation for their efforts!
+
+    local zigDefaultBuildFlags="-Drelease-safe -Dcpu=baseline"
+
+    zigBuildFlags="$zigDefaultBuildFlags $zigBuildFlags"
+
+    echo "zig build flags: $zigBuildFlags ${zigBuildFlagsArray[@]}"
+
+    zig build $zigBuildFlags "${zigBuildFlagsArray[@]}"
+
+    runHook postBuild
+}
+
+function zigCheckPhase {
+    runHook preCheck
+
+    echo "zig check flags: $zigCheckFlags ${zigCheckFlagsArray[@]}"
+
+    zig build test $zigCheckFlags "${zigCheckFlagsArray[@]}"
+
+    runHook postCheck
+}
+
+function zigInstallPhase {
+    runHook preInstall
+
+    if [ -z "${dontAddPrefix-}" ]; then
+        zigInstallFlags="$zigInstallFlags --prefix $prefix"
+    fi
+
+    echo "zig install flags: $zigBuildFlags ${zigBuildFlagsArray[@]} $zigInstallFlags ${zigInstallFlagsArray[@]}"
+
+    zig build install $zigBuildFlags "${zigBuildFlagsArray[@]}" $zigInstallFlags "${zigInstallFlagsArray[@]}"
+
+    runHook postInstall
+}
+
+addEnvHooks "$targetOffset" zigSetGlobalCacheDir
+
+if [ -z "${dontUseZigBuild-}" -a -z "${buildPhase-}" ]; then
```

Comment:
just a suggestion, I feel like this could be shorter, same suggestion for the other `dontUse` options

```suggestion
if [ -z "${dontZigBuild-}" -a -z "${buildPhase-}" ]; then
```

Reply:
Other similar codes name their variables the same. Indeed my inspirations (meson, ninja, cmake) use vars like `dontUseMesonConfigure`, `dontUseNinjaCheck`...

It's an unwritten standard I want to keep. Also, I like `longButDescriptiveCamelCaseVariable`s.